Provide System.Memory.Data package readme (#93516)
dotnet/runtime: src/libraries/System.Memory.Data/src/PACKAGE.md

<!-- A description of the package and where one can find more documentation -->

> System.Memory.Data introduces the `BinaryData` type, a lightweight abstraction for a byte payload, aiding in the effortless conversion among string, bytes, and stream.

Reviewer: ```suggestion
System.Memory.Data introduces the `BinaryData` type, a lightweight abstraction for a byte payload. It makes it easy to convert between string, bytes, and stream.
```

> This abstraction proves beneficial for simplifying API surface areas by exposing a single type instead of numerous overloads or properties.

Reviewer: ```suggestion
This abstraction can simplify API surface by exposing a single type instead of numerous overloads or properties.
```

// there is also a ToMemory method that gives access to the ReadOnlyMemory.
rom = data.ToMemory();

// and a ToArray method that converts into a byte array.
byte[] array = data.ToArray();
```

To/From stream:

```csharp
var bytes = Encoding.UTF8.GetBytes("some data");
Stream stream = new MemoryStream(bytes);
var data = BinaryData.FromStream(stream);

// Calling ToStream will give back a stream that is backed by ReadOnlyMemory, so it is not writable.
stream = data.ToStream();
Console.WriteLine(stream.CanWrite); // prints false
```

`BinaryData` also can be used to integrate with `ObjectSerializer`.
By default, the `JsonObjectSerializer` will be used, but any serializer deriving from `ObjectSerializer` can be used.

```csharp
var model = new CustomModel
{
    A = "some text",
    B = 5,
    C = true
};

var data = BinaryData.FromObjectAsJson(model);
model = data.ToObjectFromJson<CustomModel>();
```

## Main Types

<!-- The main types provided in this library -->

The main types provided by this library are:

* `System.BinaryData`

## Additional Documentation

<!-- Links to further documentation. Remove conceptual documentation if not available for the library. -->

> * [API documentation](https://learn.microsoft.com/en-us/dotnet/api/system.binarydata)

Reviewer 2: ```suggestion
* [API documentation](https://learn.microsoft.com/dotnet/api/system.binarydata)
```
